pennylane_qulacs/qulacs_device.py

```diff
@@ -192,6 +173,23 @@ def apply_operations(self, operations):
             else:
                 self._apply_gate(op)
 
+    def _expand_state(self, state_vector, wires):
+        """Expands state vector to more wires"""
+        basis_states = np.array(list(it.product([0, 1], repeat=len(wires))))
+
+        # get basis states to alter on full set of qubits
+        unravelled_indices = np.zeros((2 ** len(wires), self.num_wires), dtype=int)
+        unravelled_indices[:, wires] = basis_states
+
+        # get indices for which the state is changed to input state vector elements
+        ravelled_indices = np.ravel_multi_index(unravelled_indices.T, [2] * self.num_wires)
+
+        state_vector = self._scatter(ravelled_indices, state_vector, [2 ** self.num_wires])
+        state_vector = self._reshape(state_vector, [2] * self.num_wires)
+        state_vector = self._asarray(state_vector, dtype=self.C_DTYPE)
+
+        return state_vector.flatten()
```

Comment:
Tested this locally on a small example, works well!

```pycon
>>> sv = np.array([1, 2, 3, 4])
>>> _expand_state(sv, [2, 0], num_wires=3)
array([1., 3., 0., 0., 2., 4., 0., 0.])
```

(I modified it slightly to make it a function where you need to pass the number of device wires manually).

Reply:
Nice! 🎉